src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/framework/core/Node.java

```diff
@@ -187,4 +194,20 @@ public void setLocalFlowUnit(T localFlowUnit) {
   public void readRcaConf(RcaConf conf) {
     return;
   }
+
+  public void setAppContext(final AppContext appContext) {
+    this.appContext = appContext;
+  }
+
+  public InstanceDetails getInstanceDetails() {
+    return this.appContext.getMyInstanceDetails();
```

Comment:
do wen need to check whether appContext is null here to avoid null pointer exception ?

Reply:
Thanks a lot @rguo-aws Will add it